### PR TITLE
Doc: Adding descriptions and missing props to treepicker example

### PR DESCRIPTION
**Pull request by ChaoLiangSuper**
Adding missing props to treepicker example

## Description

Adding missing props to treepicker example
And adding descriptions to those props
The mismatch proptypes are related to this issue https://github.com/Adslot/adslot-ui/issues/689

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Screenshots (if appropriate):
screen shot 2018-02-09 at 2 35 15 pm

screen shot 2018-02-09 at 2 35 28 pm

## Check-list:

- [x] I have read the [Contributing](CONTRIBUTING.md) document.
- [x] I've thought about and labelled my PR/commit message appropriately.
- [x] If this PR introduces breaking changes I've described the impact and migration path for existing applications.
- [ ] CI is green (coverage, linting, tests).
- [x] I have updated the documentation accordingly.
- [ ] I've two LGTMs/Approvals.
- [ ] I've fixed or replied to all my code-review comments.
- [x] I've manually tested with a buddy.
- [x] I've squashed my commits into one.

### CODE REVIEW

File: docs/examples/TreePickerExample.jsx [hunk 1]
```diff
@@ -217,13 +229,13 @@ const exampleProps = {
       propType: 'searchOnChange',
       type: 'bool',
       defaultValue: <pre>true</pre>,
-      note: 'Serach for item on onChange trigger when the value is true',
+      note: 'Search for item on onChange trigger when the value is true',
```
Comment (nimishjha) [inline docs/examples/TreePickerExample.jsx]: "When true, search is triggered as soon as the user types in the search field"

File: docs/examples/TreePickerExample.jsx [hunk 2]
```diff
@@ -217,13 +229,13 @@ const exampleProps = {
...
     },
     {
       propType: 'searchOnEnterKey',
       type: 'bool',
       defaultValue: <pre>false</pre>,
-      note: 'Serach for item on Enter key trigger when the value is true',
+      note: 'Search for item on Enter key trigger when the value is true',
```
Comment (nimishjha) [inline docs/examples/TreePickerExample.jsx]: "When true, triggers search when the user presses the Enter key"

File: docs/examples/TreePickerExample.jsx [hunk 3]
```diff
@@ -94,15 +94,22 @@ const exampleProps = {
...
+      type: 'arrayOf(string)',
+      note: 'Class Names for SplitPane component',
+    },
     {
       propType: 'breadcrumbNodes',
       type: 'arrayOf { id: string/number, label: string }',
-      note: 'returns node id',
+      note:
+        'returns node id. This prop is not required,' +
+        'but an empty array is not allowed. At least one element is required in the array.',
```
Comment (omgaz) [inline docs/examples/TreePickerExample.jsx]: back-ticks allow multi-line strings in this instance.

``` es6
note:
  `returns node id. This prop is not required,
  but an empty array is not allowed. At least one element is required in the array.`,
```

File: docs/examples/TreePickerExample.jsx [hunk 4]
```diff
@@ -114,7 +121,7 @@ const exampleProps = {
       propType: 'disabled',
       type: 'bool',
       defaultValue: <pre>false</pre>,
-      note: 'disables treepicker including serach bar',
+      note: 'disables treepicker including search bar',
```
Comment (omgaz) [inline docs/examples/TreePickerExample.jsx]: :+1:

**Comment by omgaz** [comment]
Merging :ramen:

**Pull request merged**